os/kernel/semaphore/sem_holder.c

```diff
@@ -563,9 +564,10 @@ static int sem_restoreholderprioA(FAR struct semholder_s *pholder, FAR sem_t *se
 
 static int sem_restoreholderprioB(FAR struct semholder_s *pholder, FAR sem_t *sem, FAR void *arg)
 {
-	FAR struct tcb_s *rtcb = this_task();
-	if (pholder->htcb == rtcb) {
-		(void)sem_restoreholderprio(pholder, sem, arg);
+	sem_restore_t *restore_tcbs = (sem_restore_t *)arg;
+	FAR struct tcb_s *htcb = restore_tcbs->htcb;
+	if (pholder->htcb == htcb) {
+		(void)sem_restoreholderprio(pholder, sem, restore_tcbs->stcb);
```

Comment:
any reason to make it void here and returning 1 here...

Reply:
sem_restoreholderprio always return 0, so we don't use the return of it.
And 1 also has not special reason I think.
But it is not related to this PR. I will check again and remove/change if needed with different commit. Thanks :)